src/main/java/copado/onpremise/job/OnPremiseDeploymentJob.java

```diff
@@ -76,48 +94,207 @@ public void execute() {
             gitTMP = createTemporalGitPath();
             deployZipFileTMPDir = createTemporalDeployZipPath();
 
-            GitSession git = gitService.cloneRepo(gitTMP);
+            GitSession git = gitService.cloneRepo(gitTMP, gitCredentialService.getCredentialsForMainRepository());
             DeployRequest request = downloadAllBranchesAndReadDeploymentRequest(git);
             deploymentJobId = request.getDeploymentJobId();
             Path deployZipFileTMP = copyDeployZipToTemporalDir(gitTMP, deployZipFileTMPDir, git);
             log.atInfo().log("Deploy zip file tmp: %s", deployZipFileTMP);
             validatePromoteBranch(request, gitTMP, git, deployZipFileTMP);
 
-            deployZip(request, deployZipFileTMP);
+            DeploymentResult deploymentResult = deployZip(request, deployZipFileTMP);
+            processDeploymentResult(git, request, deploymentResult);
 
-            mergeAndPushDeployment(request, git);
-
-            copadoService.updateDeploymentJobStatus(request.getDeploymentJobId(), "Success");
-
-
-        } catch (CopadoException e) {
-            log.atSevere().withCause(e).log("On premise deployment failed");
-            copadoService.updateDeploymentJobStatus(deploymentJobId, buildErrorStatus(e));
-        } catch (UnexpectedErrorFault e) {
-            log.atSevere().withCause(e).log("On premise deployment failed");
-            copadoService.updateDeploymentJobStatus(deploymentJobId, buildErrorStatus(e));
         } catch (Exception e) {
             log.atSevere().withCause(e).log("On premise deployment failed");
-            copadoService.updateDeploymentJobStatus(deploymentJobId, buildErrorStatus(e));
+            copadoService.updateDeploymentJobStatus(deploymentJobId, FAILED);
         } finally {
             pathService.safeDelete(gitTMP);
             pathService.safeDelete(deployZipFileTMPDir);
         }
     }
 
-    private void mergeAndPushDeployment(DeployRequest request, GitSession git) throws CopadoException {
+    private void processDeploymentResult(GitSession git, DeployRequest request, DeploymentResult deploymentResult) throws CopadoException {
```

Comment:
IMHO private methods accessing class components are harmful, they hide complex relationships and are not tested

Reply:
I've made as private this method because it should not be overwritten by the customer, It has specific Copado behaviour.

Even in this case, the only line that it is not tested inside this method is this one:
`copadoService.updateDeploymentJobStatus(request.getDeploymentJobId(), FAILED);`

And this line has been tested manually by our product owner.